packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable func-names */
+/* global web3 */
+const { getContractAddressesForNetwork, NetworkId: networkIDs } = require('@aztec/contract-addresses');
+const contractArtifacts = require('@aztec/contract-artifacts');
+const {
+    errors: { codes, AztecError },
+} = require('@aztec/dev-utils');
+const TruffleContract = require('@truffle/contract');
+
+const { capitaliseFirstChar } = require('./utils');
+
+class Setup {
+    /**
+     * Sets up the environment for an integration test. Specifically, it creates Truffle Contract instances
+     * from the aztec contract-artifacts; at the appropriate address in the contract-addresses package.
+     *
+     * Can be used to setup a local development testing environment for an integration test
+     *
+     * @param {Object} config - config used to modify the test environment setup
+     * @param {String} accounts - Ethereum addresses to be used in the test
+     */
+    constructor(config, accounts) {
+        this.config = config;
+        this.NETWORK = config.NETWORK;
+        this.runAdjustSupplyTests = config.runAdjustSupplyTests;
+        this.runUpgrade = config.runUpgrade;
+        this.contractsToDeploy = config.contractsToDeploy;
+
+        this.provider = web3.currentProvider;
+        this.accounts = accounts;
+
+        const networks = ['ropsten', 'rinkeby', 'kovan', 'mainnet'];
+        if (networks.includes(this.NETWORK)) {
+            console.log('Configuring integration test for', this.NETWORK, 'network');
+            this.networkId = networkIDs[capitaliseFirstChar(this.NETWORK)];
+            this.getAddresses();
```

Comment:
Minor, but from a clarity point of view, would be nice if assignment to class variables was done in constructor.
```suggestion
            this.contractAddresses = this.getAddresses();
```
with this.getAddresses() simply returning data instead of doing assignment. Would make these initialisation fns closer to getters, with no or few unexpected side-effects.

Reply:
Not relevant here, but this would mean you could expose a getter to the end user of the class, and it would perform as expected (return the data) instead of modifying the instance you're working with.

Reply:
Yep, very good point. Implemented